### DISCUSSION TITLE
Is it possible to define laser profile with data points?

### DISCUSSION THREAD

**Discussion by SenpaiMario** [epochpic/epoch#450] [Q&A]
Dear EPOCH team

Is it possible to initialize the laser with an arbitrary profile from a  binary input file? I am looking to include an arbitrary t_profile in my model. I know there is an interpolate function. Can I use it for this purpose?  Do I have to include a full array of relevant points If I want to model an arbitrary laser profile?

Best regards

**Comment by Status-Mirror** [comment]
I might be wrong, but I don't think the laser profiles support binary file reading. You can use the `interpolate` function though, described [here](https://epochpic.github.io/documentation/code_details/maths_parser.html). For a very basic triangular profile on the boundary, described by 3 point-value pairs peaking at 50fs, the syntax would look like this:

```perl
t_profile = interpolate(time, 0.0e-15, 0.0, 50.0e-15, 1.0, 100.0e-15, 0.0, 3) 
```

Hope this helps,
Stuart
